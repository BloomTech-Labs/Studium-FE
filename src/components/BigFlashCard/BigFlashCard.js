import React, {useState} from 'react';
import styled from 'styled-components';
import {Card} from 'antd';
import PropTypes from 'prop-types';

/**
 * Big Flash Card
 *
 * @component
 * @example
 * const flashCard = {question: "Question?", answer: "Answer"}
 * return (
 *  <BigFlashCard flashCard={flashCard} />
 *  )
 */
const BigFlashCard = ({flashCard}) => {
  const [position, setPosition] = useState('front');
  
  const flipCard = () => {
    
    const newPos = position === 'front' ? 'back' : 'front';
    setPosition(newPos);
  };
  
  return (
    <StyledCardContainer data-testid="StyledCardContainer"
                         style={{position: 'relative'}} onClick={flipCard}>
      <StyledCard
        position={position}
        
        style={{width: '285px', height: '421.56px'}}
      >
        <CardText>
          {position === 'front' ? flashCard.question : flashCard.answer}
        </CardText>
      </StyledCard>
    </StyledCardContainer>
  
  );
};

BigFlashCard.prototypes = {
  flashCard: PropTypes.objectOf({
    question: PropTypes.string.isRequired, answer: PropTypes.string.isRequired,
  }).isRequired,
};

const StyledCardContainer = styled.div`
  height: 390px;
  width: 283px;
  position: relative;
`;

const StyledCard = styled(Card)`
  && {
    margin: 12px auto 0 auto;
    background: ${props => props.position === 'front' ? '#F7F7F7' :
  '#1b1414c9'};
    color: ${props => (props.position === 'front' ? '#1b1414c9' : 'white')};

    border-radius: 11px;
    > .ant-card-body {
      min-height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
  transition: all 1s;
`;
const CardText = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  font-size: 20px;
  font-weight: 900;
  transition: all 1s;
`;
export default BigFlashCard;
