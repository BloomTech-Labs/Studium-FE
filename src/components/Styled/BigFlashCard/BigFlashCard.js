import React, { useState } from 'react';
import styled from 'styled-components';
import card2 from '../../../images/Card2.jpg';
import card3 from '../../../images/Card3.jpg';
import { Card } from 'antd';
import PropTypes from 'prop-types';

/**
 * Big Flash Card
 *
 * @component
 * @example
 * const flashCard = {question: "Question?", answer: "Answer"}
 * return (
 *  <BigFlashCard flashCard={flashCard} />
 *  )
 */
const BigFlashCard = ( { flashCard } ) => {
  const [ position, setPosition ] = useState( 'front' );
  
  const flipCard = () => {
    
    const newPos = position === 'front' ? 'back' : 'front';
    setPosition( newPos );
  };
  
  return (
    <StyledCardContainer data-testid = "StyledCardContainer" style={{ position: 'relative' }} onClick={flipCard}>
      <StyledCard
        position={position}
      
        style={{ width: '285px', height: '421.56px' }}
      >
        <CardText>
          { position === 'front' ? flashCard.question : flashCard.answer }
        </CardText>
      </StyledCard>
      <img
        src={ card2 }
        style={ {
          position: 'absolute',
          top: '14px',
          height: '417px',
          width: '276px',
          zIndex: -1,
          left: '5px',
        } }
        alt={ 'card' }
      />
      <img
        src={ card3 }
        style={ {
          position: 'absolute',
          top: '22px',
          left: '7px',
          height: '417px',
          width: '260px',
          zIndex: -3,
        } }
        alt={ 'card' }
      />
    </StyledCardContainer>
  
  );
};

BigFlashCard.prototypes = {
  flashCard: PropTypes.objectOf( {
    question: PropTypes.string.isRequired, answer: PropTypes.string.isRequired,
  } ).isRequired,
};

const StyledCardContainer = styled.div`
  height: 390px;
  width: 283px;
  position: relative;
`;

const StyledCard = styled( Card )`
  && {
    margin: 121px auto 0 auto;
    background: ${ props => props.position === 'front' ? '#F7F7F7' :
  '#1b1414c9' };
    color: ${ props => ( props.position === 'front' ? '#1b1414c9' : 'white' ) };

    border-radius: 11px;
    > .ant-card-body {
      min-height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }
  transition: all 1s;
`;
const CardText = styled.div`
  height: 100%;
  display: flex;
  justifycontent: center;
  font-size: 20px;
  font-weight: 900;
  transition: all 1s;
`;
export default BigFlashCard;
