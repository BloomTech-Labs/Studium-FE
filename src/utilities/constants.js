/**
 * @typedef {number} LogType
 *
 */

/**
 * @typedef {object} LOG_TYPES
 * @param {logType} INFO
 * @param {logType} WARNING
 * @param {logType} ERROR
 * @param {logType} OBJECT
 */
export const LOG_TYPES = {
  VERBOSE: 1,
  INFO: 2,
  WARNING: 4,
  ERROR: 5,
  OBJECT: 6,
  PLAIN: 3,
};

/**
 * @typedef {string} DEBUG_LOG_SIZE
 *
 */

/**
 * @typedef {object} DEBUG_LOG_SIZES
 * @param {DEBUG_LOG_SIZE} LARGE
 * @param {DEBUG_LOG_SIZE} MEDIUM
 * @param {DEBUG_LOG_SIZE} NORMAL
 * @param {DEBUG_LOG_SIZE} SMALL
 */
export const DEBUG_LOG_SIZES = {
  LARGE: 'large',
  MEDIUM: 'medium',
  NORMAL: 'normal',
  SMALL: 'small',
};

/**
 * @typedef {("APP_VIEW_DESKTOP" | "APP_VIEW_MOBILE")} AppView
 *
 */

/**
 * @type {ThemeState}
 */
export const THEME = {
  primaryColor: '#0d2545',
  primaryColorB98C4: '#b9b8c4',
  primaryColor86869A: '#86869a',
  primaryColor5C5F78: '#5C5F78',
  primaryColor36405C: '#36405c',
  primaryColor293046: '#293046',
  primaryColor353544: '#353544',
  primaryColor122543: '#122543',
  
  secondaryColor: '#A2D8C7',
  secondary4CB69F: '#4CB69F',
  secondaryColorEAF5F1: '#EAF5F1',
  secondaryColorD8EEE6: '#D8EEE6',
  secondaryColorC6E6DB: '#C6E6DB',
  secondaryColorB5DFD1: '#B5DFD1',
  secondaryColor92B2AA: '#92B2AA',
  secondaryColor798A87: '#798A87',
  
  white: '#FFFFFF',
  grayF1F2F2: '#F1F2F2',
  grayE6E7E8: '#E6E7E8',
  grayD1D3D4: '#D1D3D4',
  grayBCBEC0: '#BEBEC0',
  grayA7A9AC: '#A7A9AC',
  gray939598: '#939598',
  
  lightNavBarColor: '#E5E5E5',
  
  largeRadius: 14,
  smallRadius: 6,
  navBarTopHeight: 75,
  footerHeight: 70,
  
  navBarDark: '#0C2545',
  navBarLight: '#F6F5F3',
  
  synapsDark: '#36405C',
  synapsLight: '#FFFFFF',
  
  brainPicDark: '#164167',
  brainPicLight: '#e1ded7',
};

/**
 * @typedef {Object} ThemeState
 * @property {Color} primaryColor
 * @property {Color} primaryColorB98C4
 * @property {Color} primaryColor86869A
 * @property {Color} primaryColor5C5F78
 * @property {Color} primaryColor36405C
 * @property {Color} primaryColor293046
 * @property {Color} primaryColor353544
 *
 * @property {Color} secondaryColor
 * @property {Color} secondaryColorEAF5F1
 * @property {Color} secondaryColorD8EEE6
 * @property {Color} secondaryColorC6E6DB
 * @property {Color} secondaryColorB5DFD1
 * @property {Color} secondaryColor92B2AA
 * @property {Color} secondaryColor798A87
 *
 * @property {Color} white
 * @property {Color} grayF1F2F2
 * @property {Color} grayE6E7E8
 * @property {Color} grayD1D3D4
 * @property {Color} grayBCBEC0
 * @property {Color} grayA7A9AC
 * @property {Color} gray939598
 *
 * @property {number} largeRadius
 * @property {number} smallRadius
 * @property {number} navBarTopHeight
 * @property {number} footerHeight
 *
 * @property {string} navBarLight
 * @property {string} navBarDark
 *
 * @property {THEMING_VALUES} NAV_STYLE
 *
 * @property {Color} brainPicDark
 * @property {Color} brainPicLight
 *
 */

export const APP_VIEW_MOBILE = 'APP_VIEW_MOBILE';
export const APP_VIEW_DESKTOP = 'APP_VIEW_DESKTOP';

/**
 * @typedef {string} APP_PATH
 */

/**
 * @typedef {object.<{APP_PATH}, string>} AppPaths
 */
export const APP_PATHS = {
  SIGN_UP_PATH: '/signup',
  SIGN_IN_PATH: '/signin',
  DASHBOARD_PATH: '/dashboard',
  CREATE_DECK_PATH: '/create/deck',
  PREVIEW_DECK_PATH: '/preview',
  GAME_PATH: '/game',
  TESTING: '/test',
  LANDING_PAGE: '/',
  QUIZ_MODE: '/quiz-mode',
};

/**
 * @type Sizes
 *
 */
export const SIZES = {
  mobileS: 320,
  mobileM: 375,
  mobileL: 425,
  tablet: 768,
  laptop: 1024,
  laptopL: 1440,
  desktop: 2560,
};
/**
 * @category Utilities
 * @type {Devices}
 */
export const MEDIA_QUERIES = {
  mobileS: `(min-width: ${SIZES.mobileS}px)`,
  mobileM: `(min-width: ${SIZES.mobileM}px)`,
  mobileL: `(min-width: ${SIZES.mobileL}px)`,
  tablet: `(min-width: ${SIZES.tablet}px)`,
  laptop: `(min-width: ${SIZES.laptop}px)`,
  laptopL: `(min-width: ${SIZES.laptopL}px)`,
  desktop: `(min-width: ${SIZES.desktop}px)`,
};

/**
 * @typedef {function} Dispatch
 */

/**
 * @typedef {string} Color
 */

/**
 * @typedef {object} Sizes
 * @property {number} mobileS '320px',
 * @property {number} mobileM '375px',
 * @property {number} mobileL '425px',
 * @property {number} tablet '768px',
 * @property {number} laptop '1024px',
 * @property {number} laptopL '1440px',
 * @property {number} desktop '2560px',
 */

/**
 * @typedef {string} MediaQuery
 */

/**
 * @typedef {object} Devices
 * @property {MediaQuery} mobileS   320px
 * @property {MediaQuery} mobileM   375px
 * @property {MediaQuery} mobileL   425px
 * @property {MediaQuery} tablet    768px
 * @property {MediaQuery} laptop    1024px
 * @property {MediaQuery} laptopL   1440px
 * @property {MediaQuery} desktop   2560px
 */
