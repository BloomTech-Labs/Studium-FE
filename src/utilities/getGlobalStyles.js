import {createGlobalStyle} from 'styled-components';
import reset from 'styled-reset';

const GlobalStyle = createGlobalStyle`
${reset}
  body {
        font-family: 'Source Sans Pro', sans-serif;
  }  
`;

export const getGlobalStyles = () => GlobalStyle;
