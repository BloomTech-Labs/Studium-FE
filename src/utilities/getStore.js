import React, {useState} from 'react';
import {composeWithDevTools} from 'redux-devtools-extension';
import {createStore, applyMiddleware} from 'redux';
import Thunk from 'redux-thunk';
import {storageBackUp, logger} from '../middleWare/reduxMiddware.js';
import rootReducer from '../reducers';

/**
 * Get Store
 *
 * @category Utilities
 * @function
 * @name getStore
 * @returns Store
 *
 */
export const getStore = (initialState) => {
  let store = null;
  
  if(initialState){
    try{
      store = createStore(
        rootReducer,
        initialState,
        composeWithDevTools(applyMiddleware(storageBackUp, logger, Thunk)),
      );
      if(store){
        const state = store.getState();
        return store;
      }
    }catch(e){
      store = createStore(
        rootReducer,
        composeWithDevTools(applyMiddleware(storageBackUp, logger, Thunk)),
      );
      if(store){
        return store;
      }
    }
  }
  
  store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(storageBackUp, logger, Thunk)),
  );
  return store;
};

/**
 * @typedef {object} Store
 * @property {function} dispatch
 * @property {function} subscribe
 * @property {function} getState
 * @property {function} replaceReducer
 */
