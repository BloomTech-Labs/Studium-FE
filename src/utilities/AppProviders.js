import React from 'react';
import {Provider} from 'react-redux';
import {getStore} from './getStore.js';
import {getGlobalStyles} from './getGlobalStyles.js';
import {
  useAppHooksState, AppHooksContext, useAppHooks,
} from '../customHooks/useAppHooks.js';
import {ErrorBoundary} from '../components';
import {REDUCER_NAMES} from '../reducers';
import {SYNAPS_CONFIG} from '../synapsConfig.js';
import {
  useThemeContext, useThemeRules,
} from '../customHooks/useThemeContext.js';
import {useDimensions} from '../customHooks/useDimensions.js';
import {ThemeProvider} from 'styled-components';
import {useAppView} from '../customHooks/useAppView.js';
import {BrowserRouter as Router} from 'react-router-dom';

export const APP_PROVIDER_DEBUG_NAME = 'App Provider';
const GlobalStyles = getGlobalStyles();
const initialState = {};

if(process.env.NODE_ENV !== 'test'){
  
  REDUCER_NAMES.forEach(key => {
    try{
      
      const stateKey = JSON.parse(
        localStorage.getItem(SYNAPS_CONFIG.localStorageBasePath + key),
      );
      
      if(typeof stateKey === 'object' && stateKey !== null){
        initialState[key] = stateKey;
      }else{
      
      }
      
    }catch(e){
      
      initialState[key] = '';
    }
  });
}
let store;
if(Object.values(initialState).length >= REDUCER_NAMES.length){
  store = getStore(
    Object.values(initialState).length >= REDUCER_NAMES.length && initialState,
  );
}else{
  store = getStore(false);
}

/**
 * App Providers
 * @category Utilities
 *
 * @description Store and theme provider setup for our application.
 */
const AppProvider = props => {
  
  const {themeRules, themeState, changeTheme} = useThemeRules();
  
  return (
    <ErrorBoundary>
      <Router>
        <ThemeProvider theme={{changeTheme, themeState, ...themeRules}}>
          <Provider store={store}>
            <AfterStoreTheme {...props} />
          </Provider>
        </ThemeProvider>
      </Router>
    
    </ErrorBoundary>
  );
};

const AfterStoreTheme = props => {
  
  const {hooks, setHookVariable} = useAppHooksState();
  
  return (
    <>
      <GlobalStyles/>
      <AppHooksContext.Provider
        value={{setHookVariable, hooks}}>
        
        <AfterHooks{...props}/>
      </AppHooksContext.Provider>
    </>
  );
};

const AfterHooks = props => {
  
  useThemeContext();
  useDimensions();
  useAppView();
  const {getHooks} = useAppHooks();
  const {children, ...rest} = props;
  const newChildren = React.cloneElement(props.children, {getHooks, ...rest});
  return (
    <>
      {newChildren}
    </>
  );
};

export default AppProvider;
