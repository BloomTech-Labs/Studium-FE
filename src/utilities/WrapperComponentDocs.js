// wrapper-component.js
import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme.js';
import { StyleSheetManager } from 'styled-components';
import { getStore } from './getStore.js';

const WrapperComponentDocs = ( props ) => {
  const { frameContext } = props;
  return (
    <StyleSheetManager target={ frameContext.document.head }>
      <ThemeProvider theme={ theme }>
        <Provider store={ getStore() }>
          { props.children }
        </Provider>
      </ThemeProvider>
    </StyleSheetManager>
  );
};

export default WrapperComponentDocs;