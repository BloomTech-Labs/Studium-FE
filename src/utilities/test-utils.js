import React from 'react';
import {render} from '@testing-library/react';
import AppProvider from './AppProviders.js';
import {getStore} from './getStore.js';

const store = getStore();

/**
 * Custom Render
 * @category Utilities
 *
 * @description Custom render function to wrap our components with provides
 * they need for testing purposes.
 *
 * @param ui
 * @param options
 * @return {CustomRenderResults}
 *
 */
export const customRender = (ui, options) => render(ui, {
  wrapper: AppProvider, ...options,
});

/**
 * Get Child Nodes
 * @category Utilities
 *
 * @description Gets all the child nodes from a HtmlElment for react testing
 * library.
 *
 * @param {HTMLElement} c
 * @return {HTMLElement[]}
 */
export const getChildNodes = c => {
  console.log('inside of get next node.');
  const childNodes = [];
  if(c.hasChildNodes()){
    c.childNodes.forEach(child => {
      childNodes.push(child);
    });
  }
  return childNodes;
};

/**
 * Get Nodes By Node Type
 * @category Utilities
 *
 * @description Returns all the child HtmlElements that are of the requested
 * type.
 *
 * @param {HTMLElement} c
 * @param {string} type
 * @return {HTMLElement[]}
 */
export const getNodesByType = (c, type) => {
  const stack = [];
  const toReturn = [];
  stack.push(c);
  while(stack.length > 0){
    let node = stack.pop();
    if(node.hasChildNodes()){
      node.childNodes.forEach(child => {
        stack.push(child);
      });
    }
    if(node.nodeName === type.toUpperCase()){
      toReturn.push(node);
    }
  }
  
  return toReturn;
};

export * from '@testing-library/react';

export {store};

/**
 * @typedef {object} CustomRenderResults
 * @property {HTMLDivElement} container
 * @property {HTMLBodyElement} baseElement
 * @property {function} debug
 * @property {function} unmount
 * @property {function} rerender
 * @property {function} asFragment
 * @property {function} queryAllByLabelText
 * @property {function} getAllByLabelText
 * @property {function} queryByLabelText
 * @property {function} getAllByLabelText
 * @property {function} queryByLabelText
 * @property {function} getByLabelText
 * @property {function} findAllByLabelText
 * @property {function} findByLabelText
 * @property {function} queryByPlaceholderText
 * @property {function} queryAllByPlaceholderText
 * @property {function} getByPlaceholderText
 * @property {function} getAllByPlaceholderText
 * @property {function} findAllByPlaceholderText
 * @property {function} findByPlaceholderText
 * @property {function} queryAllByText
 * @property {function} queryByText
 * @property {function} getByText
 * @property {function} getAllByText
 * @property {function} findAllByText
 * @property {function} findByText
 * @property {function} queryAllByDisplayValue
 * @property {function} queryByDisplayValue
 * @property {function} getByDisplayValue
 * @property {function} getAllByDisplayValue
 * @property {function} findAllByDisplayValue
 * @property {function} findByDisplayValue
 * @property {function} queryAllByAltText
 * @property {function} queryByAltText
 * @property {function} getByAltText
 * @property {function} getAllByAltText
 * @property {function} findAllByAltText
 * @property {function} findByAltText
 * @property {function} queryAllByTitle
 * @property {function} queryByTitle
 * @property {function} getByTitle
 * @property {function} getAllByTitle
 * @property {function} findAllByTitle
 * @property {function} findByTitle
 * @property {function} queryAllByRole
 * @property {function} queryByRole
 * @property {function} getAllByRole
 * @property {function} getByRole
 * @property {function} findAllByRole
 * @property {function} findByRole
 * @property {function} queryByTestId
 * @property {function} queryAllByTestId
 * @property {function} getByTestId
 * @property {function} getAllByTestId
 * @property {function} findAllByTestId
 * @property {function} findByTestId
 */