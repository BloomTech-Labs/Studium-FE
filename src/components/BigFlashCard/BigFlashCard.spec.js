import React from 'react';
import {customRender, fireEvent} from '../../utilities/test-utils';
import BigFlashCard from './BigFlashCard';

/**
 * Test the big flash card
 */

// test('BigFlashCard render correctly', () => {
//   const tree = renderer.create(<BigFlashCard flashCard={{question: "This is"
// + " a question", answer: "This is the answer."}} />).toJSON();
// expect(tree).toMatchSnapshot(); });

describe('Big Flash Card', () => {
  
  test('BigFlashCard render correctly', () => {
    const {container} = customRender(<BigFlashCard flashCard={{
      question: 'This is' +
        ' a question',
    }}/>);
    expect(container).toMatchSnapshot();
  });
  
  test('position front', () => {
    const {container, getByText} = customRender(<BigFlashCard
      flashCard={{question: 'question', answer: 'answer'}}/>);
    expect(getByText('question')).toBeInTheDocument();
  });
  test('position back', () => {
    const {container, getByTestId, queryByText, debug} = customRender(
      <BigFlashCard
        flashCard={{question: 'question', answer: 'answer'}}/>);
    const card = getByTestId('StyledCardContainer');
    fireEvent.click(card);
    expect(queryByText('answer')).toBeInTheDocument();
    //debug(queryByText('answer'))
  });
  
});