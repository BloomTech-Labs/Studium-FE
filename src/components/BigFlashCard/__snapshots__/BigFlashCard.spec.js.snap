// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big Flash Card BigFlashCard render correctly 1`] = `
.c0 {
  height: 390px;
  width: 283px;
  position: relative;
}

.c1 {
  -webkit-transition: all 1s;
  transition: all 1s;
}

.c1.c1 {
  margin: 121px auto 0 auto;
  background: #F7F7F7;
  color: #1b1414c9;
  border-radius: 11px;
}

.c1.c1 > .ant-card-body {
  min-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 20px;
  font-weight: 900;
  -webkit-transition: all 1s;
  transition: all 1s;
}

<div>
   
  <div
    class="c0"
    data-testid="StyledCardContainer"
    style="position: relative;"
  >
    <div
      class="ant-card c1 ant-card-bordered"
      position="front"
      style="width: 285px; height: 421.56px;"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="c2"
        >
          This is a question
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
